### ISSUE TITLE
Crete New Pages

### ISSUE THREAD

**Issue by webnucleo** [ManifoldScholar/manifold#913]
In backend create page do nothing, black scren only. Manifold 1.0.0 .deb install

**Comment by zdavis** [comment]
Thanks for reporting this. We'll take a look today and see if we're able to reproduce.

**Comment by zdavis** [comment]
I'm not able to reproduce this with a clean install on an Ubuntu 16 VM. Would you mind opening your browser console, reloading the page, and checking if any errors are reported? Alternately, if this is on a public facing URL and you'd like to email me the URL (see my github profile for my email address) and a backend login, I can take a look and see what I can find. Or, if you'd like to join our slack channel, someone on the development team will help troubleshoot this (https://manifold-slackin.herokuapp.com/).

Thank you so much for taking the time to help us identify the problem. We want to make the installation experience as simple and straight forward as possible!

**Comment by webnucleo** [comment]
Hi Zach


Thanks for the help.

You can log in to http://foliobooks.org
user: info@webnucleo.net
password: webnucleo

If you need I can give to you the login to connect via SSH to webserver.



Zach Davis wrote:
> Alternately, if this is on a public facing URL and you'd like to email 
> me the URL (see my github profile for my email address) and a backend 
> login

-- 
Cumprimentos

Carlos Ferreira

-- -- -- -- -- -- -- -- -- -
www.webnucleo.com <http://www.webnucleo.com>
Tm: 962 438 386
-- -- -- -- -- -- -- -- -- -

**Comment by zdavis** [comment]
Thanks! We'll investigate this further today and let you know what we find.

**Comment by zdavis** [comment]
Can you tell me what OS / Browser you're seeing this error in?

**Comment by webnucleo** [comment]
In version 0.4.0 no problem to create pages.
I do a clean install in my server (Ubuntu server 16.0.4 LTS) and install 
1.0.0

My "client" operating system is Mac OS 10.11.6 (El Capitan). I test 
Manifold in (last versions) Safari+Chrome+Firefox.

In Safari and Firefox activate Web Inspector and don't see any errors 
reported. Strange!?




Zach Davis wrote:
> Can you tell me what OS / Browser you're seeing this error in?

-- 
Cumprimentos

Carlos Ferreira

-- -- -- -- -- -- -- -- -- -
www.webnucleo.com <http://www.webnucleo.com>
Tm: 962 438 386
-- -- -- -- -- -- -- -- -- -

**Comment by zdavis** [comment]
I misunderstood the problem. We're able to reproduce it now and are working on a fix. We'll release a 1.0.1 package for this, probably later today or tomorrow. We'll also update the docs with some upgrade instructions. Thanks for noticing this and bringing it to our attention! And thanks for bearing with us during these growing pains :)

**Comment by zdavis** [comment]
This has been fixed in 1.0.1, which has been released. See https://manifoldapp.org/docs/installing/ubuntu/upgrading for upgrade steps. Let us know if you run into problems.

Thanks again for reporting the issue and helping us troubleshoot! 

best,
Zach

**Comment by webnucleo** [comment]
Hi Zack

Please consider this situation:

Applied the instructions for the update via .deb package
When I access the url of the installation, error 404 Not Found for all 
pages.
I checked the manifold.rb file and the installation domain is correct.
Thinking that something could have gone wrong, I carried out a new 
installation, in a new image of ubuntu server 16.04.
I installed the manifold version 1.0.0, tested it and everything 
functional (of course, the bug in the creation of pages was kept). I 
then left for the update, after the instructions, the access problem 
remained, just as it had happened in the previous update.

I attach a file (long) output in terminal : Upgrade_Manifold_Output.txt
Thanks you for your help


carlos

Zach Davis wrote:
> This has been fixed in 1.0.1, which has been released. See 
> https://manifoldapp.org/docs/installing/ubuntu/upgrading for upgrade 
> steps. Let us know if you run into problems.

-- 


root@vps504592:~# cd ~
root@vps504592:~# curl -O https://storage.googleapis.com/manifold-dist/debian/manifold_1.0.1-1_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  305M  100  305M    0     0  3244k      0  0:01:36  0:01:36 --:--:-- 2293k
root@vps504592:~# curl -O https://storage.googleapis.com/manifold-dist/debian/manifold_1.0.0-1_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  305M  100  305M    0     0  4932k      0  0:01:03  0:01:03 --:--:-- 6688k
root@vps504592:~# ls
manifold_1.0.0-1_amd64.deb  manifold_1.0.1-1_amd64.deb
root@vps504592:~# dpkg -i manifold_1.0.0-1_amd64.deb
Selecting previously unselected package manifold.
(Reading database ... 53992 files and directories currently installed.)
Preparing to unpack manifold_1.0.0-1_amd64.deb ...
You're about to install manifold!
Unpacking manifold (1.0.0-1) ...
Setting up manifold (1.0.0-1) ...


Manifold Scholarship


manifold: Thank you for installing Manifold!
manifold: To configure and start Manifold, RUN THE FOLLOWING COMMAND:

sudo manifold-ctl reconfigure

manifold: Manifold should be reachable at http://vps504592.ovh.net
manifold: Otherwise configure Manifold for your system by editing /etc/manifold/manifold.rb file
manifold: And running reconfigure again.
manifold: 
root@vps504592:~# nano /etc/manifold/manifold.rb
root@vps504592:~# /usr/local/bin/manifold-ctl reconfigure
Starting Chef Client, version 12.12.15
resolving cookbooks for run list: ["manifold::default"]
W, [2018-04-04T23:39:22.023924 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.024216 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.024342 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.024457 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.029859 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.030007 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.030110 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.030192 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.138131 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.138352 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.138471 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
W, [2018-04-04T23:39:22.138585 #1973]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#default defined in Hash. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
Synchronizing Cookbooks:
  - package (0.0.0)
  - manifold (0.1.0)
  - runit (0.14.2)
Installing Cookbook Gems:
Compiling Cookbooks...
Recipe: manifold::default
  * directory[/etc/manifold] action create
    - change mode from '0755' to '0775'
  Converging 306 resources
  * directory[/etc/manifold] action create (up to date)
  * directory[Create /var/opt/manifold/etc] action create
    - create new directory /var/opt/manifold/etc
    - change mode from '' to '0755'
    - change owner from '' to 'root'
    - change group from '' to 'root'
  * directory[/opt/manifold/embedded/etc] action create
    - change mode from '0775' to '0755'
Recipe: manifold::web-server
  * group[Webserver user and group] action create
    - create group manifold-www
  * user[Webserver user and group] action create
    - create user manifold-www
Recipe: manifold::users
  * group[Manifold user and group] action create
    - create group manifold
  * user[Manifold user and group] action create
    - create user manifold
  * directory[/var/opt/manifold] action create
    - change owner from 'root' to 'manifold'
    - change group from 'root' to 'manifold'
Recipe: manifold::manifold-api
  * directory[/var/log/manifold] action create
    - create new directory /var/log/manifold
    - change mode from '' to '0755'
    - change owner from '' to 'manifold'
  * ruby_block[directory resource: /var/opt/manifold/api/shared] action run
    - execute the ruby block directory resource: /var/opt/manifold/api/shared
  * ruby_block[directory resource: /var/opt/manifold/api/uploads] action run
    - execute the ruby block directory resource: /var/opt/manifold/api/uploads
  * directory[create /var/opt/manifold/api/etc] action create
    - create new directory /var/opt/manifold/api/etc
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /opt/manifold/etc/manifold/api] action create
    - create new directory /opt/manifold/etc/manifold/api
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /var/opt/manifold/api/working] action create
    - create new directory /var/opt/manifold/api/working
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /var/opt/manifold/api/tmp] action create
    - create new directory /var/opt/manifold/api/tmp
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /var/opt/manifold/api/upgrade-status] action create
    - create new directory /var/opt/manifold/api/upgrade-status
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /var/log/manifold/api] action create
    - create new directory /var/log/manifold/api
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[create /var/opt/manifold/api/keys] action create
    - create new directory /var/opt/manifold/api/keys
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[/var/opt/manifold/backups] action create
    - create new directory /var/opt/manifold/backups
    - change mode from '' to '0700'
    - change owner from '' to 'manifold'
  * directory[/var/opt/manifold/api] action create
    - change owner from 'root' to 'manifold'
  * template[/opt/manifold/etc/manifold/api/manifold-api-rc] action create
    - create new file /opt/manifold/etc/manifold/api/manifold-api-rc
    - update content in file /opt/manifold/etc/manifold/api/manifold-api-rc from none to 869fdb
    --- /opt/manifold/etc/manifold/api/manifold-api-rc	2018-04-04 23:39:23.225633274 +0200
    +++ /opt/manifold/etc/manifold/api/.chef-manifold-api-rc20180404-1973-emcxk2018-04-04 23:39:23.225633274 +0200
    @@ -1 +1,2 @@
    +manifold_user='manifold'
  * templatesymlink[Create a database.yml and create a symlink to Rails root] action create
    * template[/var/opt/manifold/api/etc/database.yml] action create
      - create new file /var/opt/manifold/api/etc/database.yml
      - update content in file /var/opt/manifold/api/etc/database.yml from none to e68a49
      --- /var/opt/manifold/api/etc/database.yml	2018-04-04 23:39:23.237633226 +0200
      +++ /var/opt/manifold/api/etc/.chef-database.yml20180404-1973-5muhpa	2018-04-04 23:39:23.237633226 +0200
      @@ -1 +1,19 @@
      +# This file is managed by manifold-ctl. Manual changes will be
      +# erased! To change the contents below, edit /etc/manifold/manifold.rb
      +# and run `sudo manifold-ctl reconfigure`.
      +
      +production:
      +  adapter: postgresql
      +  encoding: unicode
      +  collation: 
      +  database: manifold_production
      +  pool: 10
      +  username: 'manifold'
      +  password: 
      +  host: '/var/opt/manifold/postgresql'
      +  port: 3034
      +  socket: 
      +  sslmode: 
      +  sslrootcert: 
      +  sslca: 
      - change mode from '' to '0644'
      - change owner from '' to 'root'
      - change group from '' to 'root'
    * link[Link /opt/manifold/embedded/src/api/config/database.yml to /var/opt/manifold/api/etc/database.yml] action create
      - unlink existing symlink to file at /opt/manifold/embedded/src/api/config/database.yml
      - create symlink at /opt/manifold/embedded/src/api/config/database.yml to /var/opt/manifold/api/etc/database.yml
  
  * directory[/opt/manifold/etc/manifold/api/env] action create
    - create new directory /opt/manifold/etc/manifold/api/env
  * file[/opt/manifold/etc/manifold/api/env/HOME] action create
    - create new file /opt/manifold/etc/manifold/api/env/HOME
    - update content in file /opt/manifold/etc/manifold/api/env/HOME from none to 919724
    --- /opt/manifold/etc/manifold/api/env/HOME	2018-04-04 23:39:23.249633178 +0200
    +++ /opt/manifold/etc/manifold/api/env/.chef-HOME20180404-1973-16h0tn	2018-04-04 23:39:23.249633178 +0200
    @@ -1 +1,2 @@
    +/var/opt/manifold
  * file[/opt/manifold/etc/manifold/api/env/RAILS_ENV] action create
    - create new file /opt/manifold/etc/manifold/api/env/RAILS_ENV
    - update content in file /opt/manifold/etc/manifold/api/env/RAILS_ENV from none to ab8e18
    --- /opt/manifold/etc/manifold/api/env/RAILS_ENV	2018-04-04 23:39:23.253633162 +0200
    +++ /opt/manifold/etc/manifold/api/env/.chef-RAILS_ENV20180404-1973-1odw35t2018-04-04 23:39:23.253633162 +0200
    @@ -1 +1,2 @@
    +production
  * file[/opt/manifold/etc/manifold/api/env/SIDEKIQ_MEMORY_KILLER_MAX_RSS] action create
    - create new file /opt/manifold/etc/manifold/api/env/SIDEKIQ_MEMORY_KILLER_MAX_RSS
    - update content in file /opt/manifold/etc/manifold/api/env/SIDEKIQ_MEMORY_KILLER_MAX_RSS from none to 6cce36
    --- /opt/manifold/etc/manifold/api/env/SIDEKIQ_MEMORY_KILLER_MAX_RSS	2018-04-04 23:39:23.253633162 +0200
    +++ /opt/manifold/etc/manifold/api/env/.chef-SIDEKIQ_MEMORY_KILLER_MAX_RSS20180404-1973-m25jvd	2018-04-04 23:39:23.253633162 +0200
    @@ -1 +1,2 @@
    +1000000
  * file[/opt/manifold/etc/manifold/api/env/BUNDLE_GEMFILE] action create
    - create new file /opt/manifold/etc/manifold/api/env/BUNDLE_GEMFILE
    - update content in file /opt/manifold/etc/manifold/api/env/BUNDLE_GEMFILE from none to e5f3a4
    --- /opt/manifold/etc/manifold/api/env/BUNDLE_GEMFILE	2018-04-04 23:39:23.257633146 +0200
    +++ /opt/manifold/etc/manifold/api/env/.chef-BUNDLE_GEMFILE20180404-1973-1ar42mt	2018-04-04 23:39:23.257633146 +0200
    @@ -1 +1,2 @@
    +/opt/manifold/embedded/src/api/Gemfile
  * file[/opt/manifold/etc/manifold/api/env/PATH] action create
    - create new file /opt/manifold/etc/manifold/api/env/PATH
    - update content in file /opt/manifold/etc/manifold/api/env/PATH from none to 16e4d8
    --- /opt/manifold/etc/manifold/api/env/PATH	2018-04-04 23:39:23.261633130 +0200
    +++ /opt/manifold/etc/manifold/api/env/.chef-PATH20180404-1973-1n5ji9g	2018-04-04 23:39:23.261633130 +0200
    @@ -1 +1,2 @@
    +/opt/manifold/bin:/opt/manifold/embedded/bin:/usr/local/bin:/bin:/usr/bin
  * link[/opt/manifold/embedded/src/api/tmp] action create
    - create symlink at /opt/manifold/embedded/src/api/tmp to /var/opt/manifold/api/tmp
    - change mode from '' to '0775'
  * link[/opt/manifold/embedded/src/api/public/system] action create
    - create symlink at /opt/manifold/embedded/src/api/public/system to /var/opt/manifold/api/uploads
    - change mode from '' to '0775'
  * link[/opt/manifold/embedded/src/api/log] action create
    - create symlink at /opt/manifold/embedded/src/api/log to /var/log/manifold/api
    - change mode from '' to '0775'
  * link[/opt/manifold/embedded/src/config/keys] action create
    - create symlink at /opt/manifold/embedded/src/config/keys to /var/opt/manifold/api/keys
    - change mode from '' to '0775'
  * link[/var/log/manifold/api/sidekiq.log] action create
    - create symlink at /var/log/manifold/api/sidekiq.log to /var/log/manifold/sidekiq/current
  * file[/opt/manifold/embedded/src/api/db/schema.rb] action create
    - change owner from 'root' to 'manifold'
  * file[/var/opt/manifold/api/RUBY_VERSION] action create
    - create new file /var/opt/manifold/api/RUBY_VERSION
    - update content in file /var/opt/manifold/api/RUBY_VERSION from none to ed5a25
    --- /var/opt/manifold/api/RUBY_VERSION	2018-04-04 23:39:23.277633066 +0200
    +++ /var/opt/manifold/api/.chef-RUBY_VERSION20180404-1973-z6y9v1	2018-04-04 23:39:23.277633066 +0200
    @@ -1 +1,2 @@
    +ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]
Recipe: manifold::add_trusted_certs
  * directory[/etc/manifold/trusted-certs] action create
    - create new directory /etc/manifold/trusted-certs
    - change mode from '' to '0755'
  * directory[/opt/manifold/embedded/ssl/certs] action create
    - change mode from '0775' to '0755'
  * file[/opt/manifold/embedded/ssl/certs/README] action create
    - create new file /opt/manifold/embedded/ssl/certs/README
    - update content in file /opt/manifold/embedded/ssl/certs/README from none to e1ca2f
    --- /opt/manifold/embedded/ssl/certs/README	2018-04-04 23:39:23.289633018 +0200
    +++ /opt/manifold/embedded/ssl/certs/.chef-README20180404-1973-1pub511	2018-04-04 23:39:23.289633018 +0200
    @@ -1 +1,4 @@
    +This directory is managed by omnibus-manifold.
    + Any file placed in this directory will be ignored
    +. Place certificates in /etc/manifold/trusted-certs.
    - change mode from '' to '0644'
  * ruby_block[Move existing certs and link to /opt/manifold/embedded/ssl/certs] action run

  * Moving existing certificates found in /opt/manifold/embedded/ssl/certs

  * Symlinking existing certificates found in /etc/manifold/trusted-certs

    - execute the ruby block Move existing certs and link to /opt/manifold/embedded/ssl/certs
Recipe: manifold::default
  * service[create a temporary puma service] action nothing (skipped due to action :nothing)
  * service[create a temporary sidekiq service] action nothing (skipped due to action :nothing)
Recipe: runit::systemd
  * directory[/usr/lib/systemd/system] action create
    - create new directory /usr/lib/systemd/system
  * cookbook_file[/usr/lib/systemd/system/manifold-runsvdir.service] action create
    - create new file /usr/lib/systemd/system/manifold-runsvdir.service
    - update content in file /usr/lib/systemd/system/manifold-runsvdir.service from none to f5040b
    --- /usr/lib/systemd/system/manifold-runsvdir.service	2018-04-04 23:39:23.353632761 +0200
    +++ /usr/lib/systemd/system/.chef-manifold-runsvdir.service20180404-1973-t7wknb	2018-04-04 23:39:23.353632761 +0200
    @@ -1 +1,11 @@
    +[Unit]
    +Description=Manifold Runit supervision process
    +After=basic.target
    +
    +[Service]
    +ExecStart=/opt/manifold/embedded/bin/runsvdir-start
    +Restart=always
    +
    +[Install]
    +WantedBy=basic.target
    - change mode from '' to '0644'
  * execute[systemctl daemon-reload] action run
    - execute systemctl daemon-reload
  * execute[systemctl enable manifold-runsvdir] action run
    [execute] Created symlink from /etc/systemd/system/basic.target.wants/manifold-runsvdir.service to /usr/lib/systemd/system/manifold-runsvdir.service.
    - execute systemctl enable manifold-runsvdir
  * execute[systemctl start manifold-runsvdir] action run
    - execute systemctl start manifold-runsvdir
  * file[/etc/systemd/system/default.target.wants/manifold-runsvdir.service] action delete (up to date)
  * execute[systemctl daemon-reload] action nothing (skipped due to action :nothing)
  * execute[systemctl enable manifold-runsvdir] action nothing (skipped due to action :nothing)
  * execute[systemctl start manifold-runsvdir] action nothing (skipped due to action :nothing)
Recipe: manifold::redis
  * group[user and group for redis] action create
    - create group manifold-redis
  * user[user and group for redis] action create
    - create user manifold-redis
  * group[Socket group] action create (up to date)
  * directory[/var/opt/manifold/redis] action create
    - create new directory /var/opt/manifold/redis
    - change mode from '' to '0750'
    - change owner from '' to 'manifold-redis'
    - change group from '' to 'manifold'
  * directory[/var/log/manifold/redis] action create
    - create new directory /var/log/manifold/redis
    - change mode from '' to '0700'
    - change owner from '' to 'manifold-redis'
  * template[/var/opt/manifold/redis/redis.conf] action create
    - create new file /var/opt/manifold/redis/redis.conf
    - update content in file /var/opt/manifold/redis/redis.conf from none to 2cd628
    --- /var/opt/manifold/redis/redis.conf	2018-04-04 23:39:23.845630789 +0200
    +++ /var/opt/manifold/redis/.chef-redis.conf20180404-1973-6nwyxt	2018-04-04 23:39:23.845630789 +0200
    @@ -1 +1,1059 @@
    +# This file is managed by manifold-ctl. Manual changes will be
    +# erased! To change the contents below, edit /etc/manifold/manifold.rb
    +# and run `sudo manifold-ctl reconfigure`.
    +
    +# Redis configuration file example.
    +#
    +# Note that in order to read the configuration file, Redis must be
    +# started with the file path as first argument:
    +#
    +# ./redis-server /path/to/redis.conf
    +
    +# Note on units: when memory size is needed, it is possible to specify
    +# it in the usual form of 1k 5GB 4M and so forth:
    +#
    +# 1k => 1000 bytes
    +# 1kb => 1024 bytes
    +# 1m => 1000000 bytes
    +# 1mb => 1024*1024 bytes
    +# 1g => 1000000000 bytes
    +# 1gb => 1024*1024*1024 bytes
    +#
    +# units are case insensitive so 1GB 1Gb 1gB are all the same.
    +
    +################################## INCLUDES ###################################
    +
    +# Include one or more other config files here.  This is useful if you
    +# have a standard template that goes to all Redis servers but also need
    +# to customize a few per-server settings.  Include files can include
    +# other files, so use this wisely.
    +#
    +# Notice option "include" won't be rewritten by command "CONFIG REWRITE"
    +# from admin or Redis Sentinel. Since Redis always uses the last processed
    +# line as value of a configuration directive, you'd better put includes
    +# at the beginning of this file to avoid overwriting config change at runtime.
    +#
    +# If instead you are interested in using includes to override configuration
    +# options, it is better to use include as the last line.
    +#
    +# include /path/to/local.conf
    +# include /path/to/other.conf
    +
    +################################## NETWORK #####################################
    +
    +# By default, if no "bind" configuration directive is specified, Redis listens
    +# for connections from all the network interfaces available on the server.
    +# It is possible to listen to just one or multiple selected interfaces using
    +# the "bind" configuration directive, followed by one or more IP addresses.
    +#
    +# Examples:
    +#
    +# bind 192.168.1.100 10.0.0.1
    +# bind 127.0.0.1 ::1
    +#
    +# ~~~ WARNING ~~~ If the computer running Redis is directly exposed to the
    +# internet, binding to all the interfaces is dangerous and will expose the
    +# instance to everybody on the internet. So by default we uncomment the
    +# following bind directive, that will force Redis to listen only into
    +# the IPv4 lookback interface address (this means Redis will be able to
    +# accept connections only from clients running into the same computer it
    +# is running).
    +#
    +# IF YOU ARE SURE YOU WANT YOUR INSTANCE TO LISTEN TO ALL THE INTERFACES
    +# JUST COMMENT THE FOLLOWING LINE.
    +# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    +bind 127.0.0.1
    +
    +# Protected mode is a layer of security protection, in order to avoid that
    +# Redis instances left open on the internet are accessed and exploited.
    +#
    +# When protected mode is on and if:
    +#
    +# 1) The server is not binding explicitly to a set of addresses using the
    +#    "bind" directive.
    +# 2) No password is configured.
    +#
    +# The server only accepts connections from clients connecting from the
    +# IPv4 and IPv6 loopback addresses 127.0.0.1 and ::1, and from Unix domain
    +# sockets.
    +#
    +# By default protected mode is enabled. You should disable it only if
    +# you are sure you want clients from other hosts to connect to Redis
    +# even if no authentication is configured, nor a specific set of interfaces
    +# are explicitly listed using the "bind" directive.
    +# protected-mode yes
    +
    +# Accept connections on the specified port, default is 6379 (IANA #815344).
    +# If port 0 is specified Redis will not listen on a TCP socket.
    +port 3035
    +
    +# TCP listen() backlog.
    +#
    +# In high requests-per-second environments you need an high backlog in order
    +# to avoid slow clients connections issues. Note that the Linux kernel
    +# will silently truncate it to the value of /proc/sys/net/core/somaxconn so
    +# make sure to raise both the value of somaxconn and tcp_max_syn_backlog
    +# in order to get the desired effect.
    +# tcp-backlog 511
    +
    +# Unix socket.
    +#
    +# Specify the path for the Unix socket that will be used to listen for
    +# incoming connections. There is no default, so Redis will not listen
    +# on a unix socket when not specified.
    +#
    +unixsocket /var/opt/manifold/redis/redis.socket
    +unixsocketperm 777
    +
    +# Close the connection after a client is idle for N seconds (0 to disable)
    +timeout 60
    +
    +# TCP keepalive.
    +#
    +# If non-zero, use SO_KEEPALIVE to send TCP ACKs to clients in absence
    +# of communication. This is useful for two reasons:
    +#
    +# 1) Detect dead peers.
    +# 2) Take the connection alive from the point of view of network
    +#    equipment in the middle.
    +#
    +# On Linux, the specified value (in seconds) is the period used to send ACKs.
    +# Note that to close the connection the double of the time is needed.
    +# On other kernels the period depends on the kernel configuration.
    +#
    +# A reasonable value for this option is 300 seconds, which is the new
    +# Redis default starting with Redis 3.2.1.
    +tcp-keepalive 300
    +
    +################################# GENERAL #####################################
    +
    +# By default Redis does not run as a daemon. Use 'yes' if you need it.
    +# Note that Redis will write a pid file in /var/run/redis.pid when daemonized.
    +daemonize no
    +
    +# If you run Redis from upstart or systemd, Redis can interact with your
    +# supervision tree. Options:
    +#   supervised no      - no supervision interaction
    +#   supervised upstart - signal upstart by putting Redis into SIGSTOP mode
    +#   supervised systemd - signal systemd by writing READY=1 to $NOTIFY_SOCKET
    +#   supervised auto    - detect upstart or systemd method based on
    +#                        UPSTART_JOB or NOTIFY_SOCKET environment variables
    +# Note: these supervision methods only signal "process is ready."
    +#       They do not enable continuous liveness pings back to your supervisor.
    +# supervised no
    +
    +# If a pid file is specified, Redis writes it where specified at startup
    +# and removes it at exit.
    +#
    +# When the server runs non daemonized, no pid file is created if none is
    +# specified in the configuration. When the server is daemonized, the pid file
    +# is used even if not specified, defaulting to "/var/run/redis.pid".
    +#
    +# Creating a pid file is best effort: if Redis is not able to create it
    +# nothing bad happens, the server will start and run normally.
    +pidfile "/var/run/redis_3035.pid"
    +
    +# Specify the server verbosity level.
    +# This can be one of:
    +# debug (a lot of information, useful for development/testing)
    +# verbose (many rarely useful info, but not a mess like the debug level)
    +# notice (moderately verbose, what you want in production probably)
    +# warning (only very important / critical messages are logged)
    +loglevel notice
    +
    +# Specify the log file name. Also the empty string can be used to force
    +# Redis to log on the standard output. Note that if you use standard
    +# output for logging but daemonize, logs will be sent to /dev/null
    +logfile ""
    +
    +# To enable logging to the system logger, just set 'syslog-enabled' to yes,
    +# and optionally update the other syslog parameters to suit your needs.
    +# syslog-enabled no
    +
    +# Specify the syslog identity.
    +# syslog-ident redis
    +
    +# Specify the syslog facility. Must be USER or between LOCAL0-LOCAL7.
    +# syslog-facility local0
    +
    +# Set the number of databases. The default database is DB 0, you can select
    +# a different one on a per-connection basis using SELECT <dbid> where
    +  # dbid is a number between 0 and 'databases'-1
    +  databases 16
    +
    +  ################################ SNAPSHOTTING  ################################
    +  #
    +  # Save the DB on disk:
    +  #
    +  #   save <seconds> <changes>
    +  #
    +  #   Will save the DB if both the given number of seconds and the given
    +  #   number of write operations against the DB occurred.
    +  #
    +  #   In the example below the behaviour will be to save:
    +  #   after 900 sec (15 min) if at least 1 key changed
    +  #   after 300 sec (5 min) if at least 10 keys changed
    +  #   after 60 sec if at least 10000 keys changed
    +  #
    +  #   Note: you can disable saving completely by commenting out all "save" lines.
    +  #
    +  #   It is also possible to remove all the previously configured save
    +  #   points by adding a save directive with a single empty string argument
    +  #   like in the following example:
    +  #
    +  #   save ""
    +
    +  save 900 1
    +  save 300 10
    +  save 60 10000
    +
    +  # By default Redis will stop accepting writes if RDB snapshots are enabled
    +  # (at least one save point) and the latest background save failed.
    +  # This will make the user aware (in a hard way) that data is not persisting
    +  # on disk properly, otherwise chances are that no one will notice and some
    +  # disaster will happen.
    +  #
    +  # If the background saving process will start working again Redis will
    +  # automatically allow writes again.
    +  #
    +  # However if you have setup your proper monitoring of the Redis server
    +  # and persistence, you may want to disable this feature so that Redis will
    +  # continue to work as usual even if there are problems with disk,
    +  # permissions, and so forth.
    +  stop-writes-on-bgsave-error yes
    +
    +  # Compress string objects using LZF when dump .rdb databases?
    +  # For default that's set to 'yes' as it's almost always a win.
    +  # If you want to save some CPU in the saving child set it to 'no' but
    +  # the dataset will likely be bigger if you have compressible values or keys.
    +  rdbcompression yes
    +
    +  # Since version 5 of RDB a CRC64 checksum is placed at the end of the file.
    +  # This makes the format more resistant to corruption but there is a performance
    +  # hit to pay (around 10%) when saving and loading RDB files, so you can disable it
    +  # for maximum performances.
    +  #
    +  # RDB files created with checksum disabled have a checksum of zero that will
    +  # tell the loading code to skip the check.
    +  rdbchecksum yes
    +
    +  # The filename where to dump the DB
    +  dbfilename "dump.rdb"
    +
    +  # The working directory.
    +  #
    +  # The DB will be written inside this directory, with the filename specified
    +  # above using the 'dbfilename' configuration directive.
    +  #
    +  # The Append Only File will also be created inside this directory.
    +  #
    +  # Note that you must specify a directory here, not a file name.
    +  dir "/var/opt/manifold/redis"
    +
    +  ################################# REPLICATION #################################
    +
    +  # Master-Slave replication. Use slaveof to make a Redis instance a copy of
    +  # another Redis server. A few things to understand ASAP about Redis replication.
    +  #
    +  # 1) Redis replication is asynchronous, but you can configure a master to
    +  #    stop accepting writes if it appears to be not connected with at least
    +  #    a given number of slaves.
    +  # 2) Redis slaves are able to perform a partial resynchronization with the
    +  #    master if the replication link is lost for a relatively small amount of
    +  #    time. You may want to configure the replication backlog size (see the next
    +  #    sections of this file) with a sensible value depending on your needs.
    +  # 3) Replication is automatic and does not need user intervention. After a
    +  #    network partition slaves automatically try to reconnect to masters
    +  #    and resynchronize with them.
    +  #
    +  # slaveof <masterip> <masterport>
    +  
    +
    +  # If the master is password protected (using the "requirepass" configuration
    +  # directive below) it is possible to tell the slave to authenticate before
    +  # starting the replication synchronization process, otherwise the master will
    +  # refuse the slave request.
    +  #
    +  # masterauth <master-password>
    +  
    +
    +  # When a slave loses its connection with the master, or when the replication
    +  # is still in progress, the slave can act in two different ways:
    +  #
    +  # 1) if slave-serve-stale-data is set to 'yes' (the default) the slave will
    +  #    still reply to client requests, possibly with out of date data, or the
    +  #    data set may just be empty if this is the first synchronization.
    +  #
    +  # 2) if slave-serve-stale-data is set to 'no' the slave will reply with
    +  #    an error "SYNC with master in progress" to all the kind of commands
    +  #    but to INFO and SLAVEOF.
    +  #
    +  slave-serve-stale-data yes
    +
    +  # You can configure a slave instance to accept writes or not. Writing against
    +  # a slave instance may be useful to store some ephemeral data (because data
    +  # written on a slave will be easily deleted after resync with the master) but
    +  # may also cause problems if clients are writing to it because of a
    +  # misconfiguration.
    +  #
    +  # Since Redis 2.6 by default slaves are read-only.
    +  #
    +  # Note: read only slaves are not designed to be exposed to untrusted clients
    +  # on the internet. It's just a protection layer against misuse of the instance.
    +  # Still a read only slave exports by default all the administrative commands
    +  # such as CONFIG, DEBUG, and so forth. To a limited extent you can improve
    +  # security of read only slaves using 'rename-command' to shadow all the
    +  # administrative / dangerous commands.
    +  slave-read-only yes
    +
    +  # Replication SYNC strategy: disk or socket.
    +  #
    +  # -------------------------------------------------------
    +  # WARNING: DISKLESS REPLICATION IS EXPERIMENTAL CURRENTLY
    +  # -------------------------------------------------------
    +  #
    +  # New slaves and reconnecting slaves that are not able to continue the replication
    +  # process just receiving differences, need to do what is called a "full
    +  # synchronization". An RDB file is transmitted from the master to the slaves.
    +  # The transmission can happen in two different ways:
    +  #
    +  # 1) Disk-backed: The Redis master creates a new process that writes the RDB
    +  #                 file on disk. Later the file is transferred by the parent
    +  #                 process to the slaves incrementally.
    +  # 2) Diskless: The Redis master creates a new process that directly writes the
    +  #              RDB file to slave sockets, without touching the disk at all.
    +  #
    +  # With disk-backed replication, while the RDB file is generated, more slaves
    +  # can be queued and served with the RDB file as soon as the current child producing
    +  # the RDB file finishes its work. With diskless replication instead once
    +  # the transfer starts, new slaves arriving will be queued and a new transfer
    +  # will start when the current one terminates.
    +  #
    +  # When diskless replication is used, the master waits a configurable amount of
    +  # time (in seconds) before starting the transfer in the hope that multiple slaves
    +  # will arrive and the transfer can be parallelized.
    +  #
    +  # With slow disks and fast (large bandwidth) networks, diskless replication
    +  # works better.
    +  # repl-diskless-sync no
    +
    +  # When diskless replication is enabled, it is possible to configure the delay
    +  # the server waits in order to spawn the child that transfers the RDB via socket
    +  # to the slaves.
    +  #
    +  # This is important since once the transfer starts, it is not possible to serve
    +  # new slaves arriving, that will be queued for the next RDB transfer, so the server
    +  # waits a delay in order to let more slaves arrive.
    +  #
    +  # The delay is specified in seconds, and by default is 5 seconds. To disable
    +  # it entirely just set it to 0 seconds and the transfer will start ASAP.
    +  # repl-diskless-sync-delay 5
    +
    +  # Slaves send PINGs to server in a predefined interval. It's possible to change
    +  # this interval with the repl_ping_slave_period option. The default value is 10
    +  # seconds.
    +  #
    +  # repl-ping-slave-period 10
    +
    +  # The following option sets the replication timeout for:
    +  #
    +  # 1) Bulk transfer I/O during SYNC, from the point of view of slave.
    +  # 2) Master timeout from the point of view of slaves (data, pings).
    +  # 3) Slave timeout from the point of view of masters (REPLCONF ACK pings).
    +  #
    +  # It is important to make sure that this value is greater than the value
    +  # specified for repl-ping-slave-period otherwise a timeout will be detected
    +  # every time there is low traffic between the master and the slave.
    +  #
    +  # repl-timeout 60
    +
    +  # Disable TCP_NODELAY on the slave socket after SYNC?
    +  #
    +  # If you select "yes" Redis will use a smaller number of TCP packets and
    +  # less bandwidth to send data to slaves. But this can add a delay for
    +  # the data to appear on the slave side, up to 40 milliseconds with
    +  # Linux kernels using a default configuration.
    +  #
    +  # If you select "no" the delay for data to appear on the slave side will
    +  # be reduced but more bandwidth will be used for replication.
    +  #
    +  # By default we optimize for low latency, but in very high traffic conditions
    +  # or when the master and slaves are many hops away, turning this to "yes" may
    +  # be a good idea.
    +  repl-disable-tcp-nodelay no
    +
    +  # Set the replication backlog size. The backlog is a buffer that accumulates
    +  # slave data when slaves are disconnected for some time, so that when a slave
    +  # wants to reconnect again, often a full resync is not needed, but a partial
    +  # resync is enough, just passing the portion of data the slave missed while
    +  # disconnected.
    +  #
    +  # The bigger the replication backlog, the longer the time the slave can be
    +  # disconnected and later be able to perform a partial resynchronization.
    +  #
    +  # The backlog is only allocated once there is at least a slave connected.
    +  #
    +  # repl-backlog-size 1mb
    +
    +  # After a master has no longer connected slaves for some time, the backlog
    +  # will be freed. The following option configures the amount of seconds that
    +  # need to elapse, starting from the time the last slave disconnected, for
    +  # the backlog buffer to be freed.
    +  #
    +  # A value of 0 means to never release the backlog.
    +  #
    +  # repl-backlog-ttl 3600
    +
    +  # The slave priority is an integer number published by Redis in the INFO output.
    +  # It is used by Redis Sentinel in order to select a slave to promote into a
    +  # master if the master is no longer working correctly.
    +  #
    +  # A slave with a low priority number is considered better for promotion, so
    +  # for instance if there are three slaves with priority 10, 100, 25 Sentinel will
    +  # pick the one with priority 10, that is the lowest.
    +  #
    +  # However a special priority of 0 marks the slave as not able to perform the
    +  # role of master, so a slave with priority of 0 will never be selected by
    +  # Redis Sentinel for promotion.
    +  #
    +  # By default the priority is 100.
    +  slave-priority 100
    +
    +  # It is possible for a master to stop accepting writes if there are less than
    +  # N slaves connected, having a lag less or equal than M seconds.
    +  #
    +  # The N slaves need to be in "online" state.
    +  #
    +  # The lag in seconds, that must be <= the specified value, is calculated from
    +  # the last ping received from the slave, that is usually sent every second.
    +  #
    +  # This option does not GUARANTEE that N replicas will accept the write, but
    +  # will limit the window of exposure for lost writes in case not enough slaves
    +  # are available, to the specified number of seconds.
    +  #
    +  # For example to require at least 3 slaves with a lag <= 10 seconds use:
    +  #
    +  # min-slaves-to-write 3
    +  # min-slaves-max-lag 10
    +  #
    +  # Setting one or the other to 0 disables the feature.
    +  #
    +  # By default min-slaves-to-write is set to 0 (feature disabled) and
    +  # min-slaves-max-lag is set to 10.
    +
    +  # A Redis master is able to list the address and port of the attached
    +  # slaves in different ways. For example the "INFO replication" section
    +  # offers this information, which is used, among other tools, by
    +  # Redis Sentinel in order to discover slave instances.
    +  # Another place where this info is available is in the output of the
    +  # "ROLE" command of a masteer.
    +  #
    +  # The listed IP and address normally reported by a slave is obtained
    +  # in the following way:
    +  #
    +  #   IP: The address is auto detected by checking the peer address
    +  #   of the socket used by the slave to connect with the master.
    +  #
    +  #   Port: The port is communicated by the slave during the replication
    +  #   handshake, and is normally the port that the slave is using to
    +  #   list for connections.
    +  #
    +  # However when port forwarding or Network Address Translation (NAT) is
    +  # used, the slave may be actually reachable via different IP and port
    +  # pairs. The following two options can be used by a slave in order to
    +  # report to its master a specific set of IP and port, so that both INFO
    +  # and ROLE will report those values.
    +  #
    +  # There is no need to use both the options if you need to override just
    +  # the port or the IP address.
    +  #
    +  # slave-announce-ip 5.5.5.5
    +  # slave-announce-port 1234
    +
    +  ################################## SECURITY ###################################
    +
    +  # Require clients to issue AUTH <PASSWORD> before processing any other
    +  # commands.  This might be useful in environments in which you do not trust
    +  # others with access to the host running redis-server.
    +  #
    +  # This should stay commented out for backward compatibility and because most
    +  # people do not need auth (e.g. they run their own servers).
    +  #
    +  # Warning: since Redis is pretty fast an outside user can try up to
    +  # 150k passwords per second against a good box. This means that you should
    +  # use a very strong password otherwise it will be very easy to break.
    +  #
    +  
    +
    +  # Command renaming.
    +  #
    +  # It is possible to change the name of dangerous commands in a shared
    +  # environment. For instance the CONFIG command may be renamed into something
    +  # hard to guess so that it will still be available for internal-use tools
    +  # but not available for general clients.
    +  #
    +  # Example:
    +  #
    +  # rename-command CONFIG b840fc02d524045429941cc15f59e41cb7be6c52
    +  #
    +  # It is also possible to completely kill a command by renaming it into
    +  # an empty string:
    +  #
    +  # rename-command CONFIG ""
    +  #
    +  # Please note that changing the name of commands that are logged into the
    +  # AOF file or transmitted to slaves may cause problems.
    +
    +  ################################### LIMITS ####################################
    +
    +  # Set the max number of connected clients at the same time. By default
    +  # this limit is set to 10000 clients, however if the Redis server is not
    +  # able to configure the process file limit to allow for the specified limit
    +  # the max number of allowed clients is set to the current file limit
    +  # minus 32 (as Redis reserves a few file descriptors for internal uses).
    +  #
    +  # Once the limit is reached Redis will close all the new connections sending
    +  # an error 'max number of clients reached'.
    +  #
    +  maxclients 10000
    +
    +  # Don't use more memory than the specified amount of bytes.
    +  # When the memory limit is reached Redis will try to remove keys
    +  # according to the eviction policy selected (see maxmemory-policy).
    +  #
    +  # If Redis can't remove keys according to the policy, or if the policy is
    +  # set to 'noeviction', Redis will start to reply with errors to commands
    +  # that would use more memory, like SET, LPUSH, and so on, and will continue
    +  # to reply to read-only commands like GET.
    +  #
    +  # This option is usually useful when using Redis as an LRU cache, or to set
    +  # a hard memory limit for an instance (using the 'noeviction' policy).
    +  #
    +  # WARNING: If you have slaves attached to an instance with maxmemory on,
    +  # the size of the output buffers needed to feed the slaves are subtracted
    +  # from the used memory count, so that network problems / resyncs will
    +  # not trigger a loop where keys are evicted, and in turn the output
    +  # buffer of slaves is full with DELs of keys evicted triggering the deletion
    +  # of more keys, and so forth until the database is completely emptied.
    +  #
    +  # In short... if you have slaves attached it is suggested that you set a lower
    +  # limit for maxmemory so that there is some free RAM on the system for slave
    +  # output buffers (but this is not needed if the policy is 'noeviction').
    +  #
    +  # maxmemory <bytes>
    +
    +  # MAXMEMORY POLICY: how Redis will select what to remove when maxmemory
    +  # is reached. You can select among five behaviors:
    +  #
    +  # volatile-lru -> remove the key with an expire set using an LRU algorithm
    +  # allkeys-lru -> remove any key according to the LRU algorithm
    +  # volatile-random -> remove a random key with an expire set
    +  # allkeys-random -> remove a random key, any key
    +  # volatile-ttl -> remove the key with the nearest expire time (minor TTL)
    +  # noeviction -> don't expire at all, just return an error on write operations
    +  #
    +  # Note: with any of the above policies, Redis will return an error on write
    +  #       operations, when there are no suitable keys for eviction.
    +  #
    +  #       At the date of writing these commands are: set setnx setex append
    +  #       incr decr rpush lpush rpushx lpushx linsert lset rpoplpush sadd
    +  #       sinter sinterstore sunion sunionstore sdiff sdiffstore zadd zincrby
    +  #       zunionstore zinterstore hset hsetnx hmset hincrby incrby decrby
    +  #       getset mset msetnx exec sort
    +  #
    +  # The default is:
    +  #
    +  # maxmemory-policy noeviction
    +
    +  # LRU and minimal TTL algorithms are not precise algorithms but approximated
    +  # algorithms (in order to save memory), so you can tune it for speed or
    +  # accuracy. For default Redis will check five keys and pick the one that was
    +  # used less recently, you can change the sample size using the following
    +  # configuration directive.
    +  #
    +  # The default of 5 produces good enough results. 10 Approximates very closely
    +  # true LRU but costs a bit more CPU. 3 is very fast but not very accurate.
    +  #
    +  # maxmemory-samples 5
    +
    +  ############################## APPEND ONLY MODE ###############################
    +
    +  # By default Redis asynchronously dumps the dataset on disk. This mode is
    +  # good enough in many applications, but an issue with the Redis process or
    +  # a power outage may result into a few minutes of writes lost (depending on
    +  # the configured save points).
    +  #
    +  # The Append Only File is an alternative persistence mode that provides
    +  # much better durability. For instance using the default data fsync policy
    +  # (see later in the config file) Redis can lose just one second of writes in a
    +  # dramatic event like a server power outage, or a single write if something
    +  # wrong with the Redis process itself happens, but the operating system is
    +  # still running correctly.
    +  #
    +  # AOF and RDB persistence can be enabled at the same time without problems.
    +  # If the AOF is enabled on startup Redis will load the AOF, that is the file
    +  # with the better durability guarantees.
    +  #
    +  # Please check http://redis.io/topics/persistence for more information.
    +
    +  appendonly no
    +
    +  # The name of the append only file (default: "appendonly.aof")
    +
    +  # appendfilename "appendonly.aof"
    +
    +  # The fsync() call tells the Operating System to actually write data on disk
    +  # instead of waiting for more data in the output buffer. Some OS will really flush
    +  # data on disk, some other OS will just try to do it ASAP.
    +  #
    +  # Redis supports three different modes:
    +  #
    +  # no: don't fsync, just let the OS flush the data when it wants. Faster.
    +  # always: fsync after every write to the append only log. Slow, Safest.
    +  # everysec: fsync only one time every second. Compromise.
    +  #
    +  # The default is "everysec", as that's usually the right compromise between
    +  # speed and data safety. It's up to you to understand if you can relax this to
    +  # "no" that will let the operating system flush the output buffer when
    +  # it wants, for better performances (but if you can live with the idea of
    +  # some data loss consider the default persistence mode that's snapshotting),
    +  # or on the contrary, use "always" that's very slow but a bit safer than
    +  # everysec.
    +  #
    +  # More details please check the following article:
    +  # http://antirez.com/post/redis-persistence-demystified.html
    +  #
    +  # If unsure, use "everysec".
    +
    +  # appendfsync always
    +  appendfsync everysec
    +  # appendfsync no
    +
    +  # When the AOF fsync policy is set to always or everysec, and a background
    +  # saving process (a background save or AOF log background rewriting) is
    +  # performing a lot of I/O against the disk, in some Linux configurations
    +  # Redis may block too long on the fsync() call. Note that there is no fix for
    +  # this currently, as even performing fsync in a different thread will block
    +  # our synchronous write(2) call.
    +  #
    +  # In order to mitigate this problem it's possible to use the following option
    +  # that will prevent fsync() from being called in the main process while a
    +  # BGSAVE or BGREWRITEAOF is in progress.
    +  #
    +  # This means that while another child is saving, the durability of Redis is
    +  # the same as "appendfsync none". In practical terms, this means that it is
    +  # possible to lose up to 30 seconds of log in the worst scenario (with the
    +  # default Linux settings).
    +  #
    +  # If you have latency problems turn this to "yes". Otherwise leave it as
    +  # "no" that is the safest pick from the point of view of durability.
    +
    +  no-appendfsync-on-rewrite no
    +
    +  # Automatic rewrite of the append only file.
    +  # Redis is able to automatically rewrite the log file implicitly calling
    +  # BGREWRITEAOF when the AOF log size grows by the specified percentage.
    +  #
    +  # This is how it works: Redis remembers the size of the AOF file after the
    +  # latest rewrite (if no rewrite has happened since the restart, the size of
    +  # the AOF at startup is used).
    +  #
    +  # This base size is compared to the current size. If the current size is
    +  # bigger than the specified percentage, the rewrite is triggered. Also
    +  # you need to specify a minimal size for the AOF file to be rewritten, this
    +  # is useful to avoid rewriting the AOF file even if the percentage increase
    +  # is reached but it is still pretty small.
    +  #
    +  # Specify a percentage of zero in order to disable the automatic AOF
    +  # rewrite feature.
    +
    +  auto-aof-rewrite-percentage 100
    +  auto-aof-rewrite-min-size 64mb
    +
    +  # An AOF file may be found to be truncated at the end during the Redis
    +  # startup process, when the AOF data gets loaded back into memory.
    +  # This may happen when the system where Redis is running
    +  # crashes, especially when an ext4 filesystem is mounted without the
    +  # data=ordered option (however this can't happen when Redis itself
    +  # crashes or aborts but the operating system still works correctly).
    +  #
    +  # Redis can either exit with an error when this happens, or load as much
    +  # data as possible (the default now) and start if the AOF file is found
    +  # to be truncated at the end. The following option controls this behavior.
    +  #
    +  # If aof-load-truncated is set to yes, a truncated AOF file is loaded and
    +  # the Redis server starts emitting a log to inform the user of the event.
    +  # Otherwise if the option is set to no, the server aborts with an error
    +  # and refuses to start. When the option is set to no, the user requires
    +  # to fix the AOF file using the "redis-check-aof" utility before to restart
    +  # the server.
    +  #
    +  # Note that if the AOF file will be found to be corrupted in the middle
    +  # the server will still exit with an error. This option only applies when
    +  # Redis will try to read more data from the AOF file but not enough bytes
    +  # will be found.
    +  # aof-load-truncated yes
    +
    +  ################################ LUA SCRIPTING  ###############################
    +
    +  # Max execution time of a Lua script in milliseconds.
    +  #
    +  # If the maximum execution time is reached Redis will log that a script is
    +  # still in execution after the maximum allowed time and will start to
    +  # reply to queries with an error.
    +  #
    +  # When a long running script exceeds the maximum execution time only the
    +  # SCRIPT KILL and SHUTDOWN NOSAVE commands are available. The first can be
    +  # used to stop a script that did not yet called write commands. The second
    +  # is the only way to shut down the server in the case a write command was
    +  # already issued by the script but the user doesn't want to wait for the natural
    +  # termination of the script.
    +  #
    +  # Set it to 0 or a negative value for unlimited execution without warnings.
    +  lua-time-limit 5000
    +
    +  ################################ REDIS CLUSTER  ###############################
    +  #
    +  # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    +  # WARNING EXPERIMENTAL: Redis Cluster is considered to be stable code, however
    +  # in order to mark it as "mature" we need to wait for a non trivial percentage
    +  # of users to deploy it in production.
    +  # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    +  #
    +  # Normal Redis instances can't be part of a Redis Cluster; only nodes that are
    +  # started as cluster nodes can. In order to start a Redis instance as a
    +  # cluster node enable the cluster support uncommenting the following:
    +  #
    +  # cluster-enabled yes
    +
    +  # Every cluster node has a cluster configuration file. This file is not
    +  # intended to be edited by hand. It is created and updated by Redis nodes.
    +  # Every Redis Cluster node requires a different cluster configuration file.
    +  # Make sure that instances running in the same system do not have
    +  # overlapping cluster configuration file names.
    +  #
    +  # cluster-config-file nodes-6379.conf
    +
    +  # Cluster node timeout is the amount of milliseconds a node must be unreachable
    +  # for it to be considered in failure state.
    +  # Most other internal time limits are multiple of the node timeout.
    +  #
    +  # cluster-node-timeout 15000
    +
    +  # A slave of a failing master will avoid to start a failover if its data
    +  # looks too old.
    +  #
    +  # There is no simple way for a slave to actually have a exact measure of
    +  # its "data age", so the following two checks are performed:
    +  #
    +  # 1) If there are multiple slaves able to failover, they exchange messages
    +  #    in order to try to give an advantage to the slave with the best
    +  #    replication offset (more data from the master processed).
    +  #    Slaves will try to get their rank by offset, and apply to the start
    +  #    of the failover a delay proportional to their rank.
    +  #
    +  # 2) Every single slave computes the time of the last interaction with
    +  #    its master. This can be the last ping or command received (if the master
    +  #    is still in the "connected" state), or the time that elapsed since the
    +  #    disconnection with the master (if the replication link is currently down).
    +  #    If the last interaction is too old, the slave will not try to failover
    +  #    at all.
    +  #
    +  # The point "2" can be tuned by user. Specifically a slave will not perform
    +  # the failover if, since the last interaction with the master, the time
    +  # elapsed is greater than:
    +  #
    +  #   (node-timeout * slave-validity-factor) + repl-ping-slave-period
    +  #
    +  # So for example if node-timeout is 30 seconds, and the slave-validity-factor
    +  # is 10, and assuming a default repl-ping-slave-period of 10 seconds, the
    +  # slave will not try to failover if it was not able to talk with the master
    +  # for longer than 310 seconds.
    +  #
    +  # A large slave-validity-factor may allow slaves with too old data to failover
    +  # a master, while a too small value may prevent the cluster from being able to
    +  # elect a slave at all.
    +  #
    +  # For maximum availability, it is possible to set the slave-validity-factor
    +  # to a value of 0, which means, that slaves will always try to failover the
    +  # master regardless of the last time they interacted with the master.
    +  # (However they'll always try to apply a delay proportional to their
    +  # offset rank).
    +  #
    +  # Zero is the only value able to guarantee that when all the partitions heal
    +  # the cluster will always be able to continue.
    +  #
    +  # cluster-slave-validity-factor 10
    +
    +  # Cluster slaves are able to migrate to orphaned masters, that are masters
    +  # that are left without working slaves. This improves the cluster ability
    +  # to resist to failures as otherwise an orphaned master can't be failed over
    +  # in case of failure if it has no working slaves.
    +  #
    +  # Slaves migrate to orphaned masters only if there are still at least a
    +  # given number of other working slaves for their old master. This number
    +  # is the "migration barrier". A migration barrier of 1 means that a slave
    +  # will migrate only if there is at least 1 other working slave for its master
    +  # and so forth. It usually reflects the number of slaves you want for every
    +  # master in your cluster.
    +  #
    +  # Default is 1 (slaves migrate only if their masters remain with at least
    +  # one slave). To disable migration just set it to a very large value.
    +  # A value of 0 can be set but is useful only for debugging and dangerous
    +  # in production.
    +  #
    +  # cluster-migration-barrier 1
    +
    +  # By default Redis Cluster nodes stop accepting queries if they detect there
    +  # is at least an hash slot uncovered (no available node is serving it).
    +  # This way if the cluster is partially down (for example a range of hash slots
    +  # are no longer covered) all the cluster becomes, eventually, unavailable.
    +  # It automatically returns available as soon as all the slots are covered again.
    +  #
    +  # However sometimes you want the subset of the cluster which is working,
    +  # to continue to accept queries for the part of the key space that is still
    +  # covered. In order to do so, just set the cluster-require-full-coverage
    +  # option to no.
    +  #
    +  # cluster-require-full-coverage yes
    +
    +  # In order to setup your cluster make sure to read the documentation
    +  # available at http://redis.io web site.
    +
    +  ################################## SLOW LOG ###################################
    +
    +  # The Redis Slow Log is a system to log queries that exceeded a specified
    +  # execution time. The execution time does not include the I/O operations
    +  # like talking with the client, sending the reply and so forth,
    +  # but just the time needed to actually execute the command (this is the only
    +  # stage of command execution where the thread is blocked and can not serve
    +  # other requests in the meantime).
    +  #
    +  # You can configure the slow log with two parameters: one tells Redis
    +  # what is the execution time, in microseconds, to exceed in order for the
    +  # command to get logged, and the other parameter is the length of the
    +  # slow log. When a new command is logged the oldest one is removed from the
    +  # queue of logged commands.
    +
    +  # The following time is expressed in microseconds, so 1000000 is equivalent
    +  # to one second. Note that a negative number disables the slow log, while
    +  # a value of zero forces the logging of every command.
    +  slowlog-log-slower-than 10000
    +
    +  # There is no limit to this length. Just be aware that it will consume memory.
    +  # You can reclaim memory used by the slow log with SLOWLOG RESET.
    +  slowlog-max-len 128
    +
    +  ################################ LATENCY MONITOR ##############################
    +
    +  # The Redis latency monitoring subsystem samples different operations
    +  # at runtime in order to collect data related to possible sources of
    +  # latency of a Redis instance.
    +  #
    +  # Via the LATENCY command this information is available to the user that can
    +  # print graphs and obtain reports.
    +  #
    +  # The system only logs operations that were performed in a time equal or
    +  # greater than the amount of milliseconds specified via the
    +  # latency-monitor-threshold configuration directive. When its value is set
    +  # to zero, the latency monitor is turned off.
    +  #
    +  # By default latency monitoring is disabled since it is mostly not needed
    +  # if you don't have latency issues, and collecting data has a performance
    +  # impact, that while very small, can be measured under big load. Latency
    +  # monitoring can easily be enabled at runtime using the command
    +  # "CONFIG SET latency-monitor-threshold <milliseconds>" if needed.
    +  # latency-monitor-threshold 0
    +
    +  ############################# EVENT NOTIFICATION ##############################
    +
    +  # Redis can notify Pub/Sub clients about events happening in the key space.
    +  # This feature is documented at http://redis.io/topics/notifications
    +  #
    +  # For instance if keyspace events notification is enabled, and a client
    +  # performs a DEL operation on key "foo" stored in the Database 0, two
    +  # messages will be published via Pub/Sub:
    +  #
    +  # PUBLISH __keyspace@0__:foo del
    +  # PUBLISH __keyevent@0__:del foo
    +  #
    +  # It is possible to select the events that Redis will notify among a set
    +  # of classes. Every class is identified by a single character:
    +  #
    +  #  K     Keyspace events, published with __keyspace@<db>__ prefix.
    +  #  E     Keyevent events, published with __keyevent@<db>__ prefix.
    +  #  g     Generic commands (non-type specific) like DEL, EXPIRE, RENAME, ...
    +  #  $     String commands
    +  #  l     List commands
    +  #  s     Set commands
    +  #  h     Hash commands
    +  #  z     Sorted set commands
    +  #  x     Expired events (events generated every time a key expires)
    +  #  e     Evicted events (events generated when a key is evicted for maxmemory)
    +  #  A     Alias for g$lshzxe, so that the "AKE" string means all the events.
    +  #
    +  #  The "notify-keyspace-events" takes as argument a string that is composed
    +  #  of zero or multiple characters. The empty string means that notifications
    +  #  are disabled.
    +  #
    +  #  Example: to enable list and generic events, from the point of view of the
    +  #           event name, use:
    +  #
    +  #  notify-keyspace-events Elg
    +  #
    +  #  Example 2: to get the stream of the expired keys subscribing to channel
    +  #             name __keyevent@0__:expired use:
    +  #
    +  #  notify-keyspace-events Ex
    +  #
    +  #  By default all notifications are disabled because most users don't need
    +  #  this feature and the feature has some overhead. Note that if you don't
    +  #  specify at least one of K or E, no events will be delivered.
    +  notify-keyspace-events ""
    +
    +  ############################### ADVANCED CONFIG ###############################
    +
    +  # Hashes ar

**Comment by webnucleo** [comment]
Hi Zack

Unbelievable, but it happened, and I can not explain what happened.
I rebooted the server, executed the commands below, again and again (2 
times), and got access.

http://foliobooks.org/


/ usr / local / bin / manifold-ctl stop
/ usr / local / bin / manifold-ctl reconfigure
/ usr / local / bin / manifold-ctl start

Maybe some problem with nginx!?

Well it looks like it's solved, thank you.

Any idea of the manifold being able to support other languages?

http://foliobooks.org/


Regards

Carlos

**Comment by susanmpls** [comment]
Hi Carlos,

Are you asking about the language on the web site's pages, or the texts that Manifold can display in the reader? Manifold's reader can already display texts in many languages. We do not have a timeline for translating the site itself, though we would love to have it available in more languages. 

cheers,
Susan

**Comment by webnucleo** [comment]
Hi Susan


I think that, at least at the beginning, it will not be necessary to 
translate the Manifold backoffice from English to another language.
The issue is, and as an example, the Manifold is being used for an 
institution to present content in (for example) Portuguese to its 
Portuguese readers. Certainly it will not be very interesting to be 
presenting some words in English that could be translated into the 
desired language.

Examples:
(EN) ->(PT)
Projects = Projects
Search = Search
Recent Activity = Recent Activity
Text = Textos
etc


susanmpls wrote:
> Are you asking about the language on the web site's pages, or the 
> texts that Manifold can display in the reader? Manifold's reader can 
> already display texts in many languages. We do not have a timeline for 
> translating the site itself, though we would love to have it available 
> in more languages.

-- 
Cumprimentos

Carlos Ferreira

-- -- -- -- -- -- -- -- -- -
www.webnucleo.com <http://www.webnucleo.com>
Tm: 962 438 386
-- -- -- -- -- -- -- -- -- -

**Comment by mkgold** [comment]
@zdavis how many of these labels could be edited manually by an admin user of a Manifold instance?
